### ISSUE TITLE
mcry gate with three control qubits cannot bind parameters

### ISSUE THREAD

**Issue by Pride-Huang** [Qiskit/qiskit-terra#10167]
### Environment

- **Qiskit Terra version**: 0.23.3
- **Python version**: 3.9
- **Operating system**: Ubuntu 18.04.6 

### What is happening?

I construct a mcry gate with three qubits as control qubits, and one qubits as target qubits. When I bind parameters, it doesn't work.
![1685085877778](https://github.com/Qiskit/qiskit-terra/assets/34713001/030b0a4f-461b-4860-a3fa-0608889ed068)

### How can we reproduce the issue?

from qiskit import QuantumCircuit, QuantumRegister, Aer
from qiskit.circuit import Parameter

qr = QuantumRegister(4)
qr_control = qr[:3]
qr_target = qr[3]

qc = QuantumCircuit(qr)
para = Parameter('input')
qc.mcry(para, qr_control, qr_target)
qc = qc.bind_parameters({para:1})
qc.draw('mpl')
![1685085877778](https://github.com/Qiskit/qiskit-terra/assets/34713001/4148c929-a1de-4ec4-a800-86a4ec9f0241)

### What should happen?

the value 1 should be binded with 'input'，but it doesn't.

**Comment by Cryoris** [comment]
This seems to be another symptom of the controlled U-gate assignment error, which will be fixed with #9118.

You can temporarily circumvent this bug by transpiling the circuit to another basis, without the `CUGate`. For example:
```python
tqc = transpile(qc, basis_gates=["h", "rz", "x", "cx"])
tqc = tqc.bind_parameters([1])
tqc.draw()
```
works as expected.

**Comment by Pride-Huang** [comment]
When will the mistake be corrected in qiskit?
